dask_geomodeling/geometry/aggregate.py

```diff
@@ -110,13 +110,164 @@ def bucketize(bboxes):
+        else:
+            thresholds = None
+
+        for frame_no, frame in enumerate(values):
+            # limit statistics to active pixels
+            active = frame != no_data_value
+            # if there is a threshold, mask the frame
+            if threshold_values is not None:
+                valid = ~np.isnan(thresholds)  # to suppress warnings
+                active[~valid] = False  # no threshold -> no aggregation
+                active[valid] &= frame[valid] >= thresholds[valid]
+
+            # if there is no single active value: do not aggregate
+            if not active.any():
+                continue
+
+            # select features that actually have data
+            # (min, max, median, and percentile cannot handle it otherwise)
+            active_labels = labels[active]
+            select_and_active = list(set(np.unique(active_labels)) & set(select))
+
+            if not select_and_active:
+                continue
+
+            agg[frame_no][select_and_active] = agg_func(
+                1 if statistic == "count" else frame[active],
+                labels=active_labels,
+                index=select_and_active,
+            )
+    return agg, list(geometries_no_cells)
+
+
+def aggregate_points(
+    points,
+    values,
+    no_data_value,
+    agg_bbox,
+    threshold_values,
+    statistic,
+):
+    """Compute aggregates for given point geometries.
+
+    The values of the raster is taken at the point coordinates.
+    
+    Args:
+      points (GeoSeries): The geometries (points) to aggregate the raster in.
+      values (ndarray): A three-dimensional raster ``(t, y, x)`` to aggregate.
+      no_data_value (number): The value in ``values`` that denotes missing data.
+      agg_bbox (tuple of 4 numbers): The bbox of ``values``.
+      threshold_values (ndarray): A threshold value per geometry.
+      statistic (str): A key in ``AggregateRaster.STATISTICS``.
+
+    Returns:
+      ndarray of dtype float32 and with shape ``(t, len(n_geometries))``
+    """
+    # Transform the points to indices
+    _, height, width = values.shape
+    gt = utils.GeoTransform.from_bbox(agg_bbox, height, width)
+    i_y, i_x = gt.get_indices(np.array([points.x.values, points.y.values]).T)
+    point_values = values[:, np.clip(i_y, 0, height - 1), np.clip(i_x, 0, width - 1)]
+
+    # if there is a threshold, mask the point values
+    active = point_values != no_data_value
+    if threshold_values is not None:
+        threshold_values = threshold_values[np.newaxis, :]  # broadcast over t
+        valid = ~np.isnan(threshold_values)  # to suppress warnings
+        active[~valid] = False  # no threshold -> no aggregation
+        active[valid] &= point_values[valid] >= threshold_values[valid]
+
+    # Convert nodata to NaN
+    agg = point_values.astype("f4")
+    agg[~active] = np.nan
+
+    # For all statistics the aggregated equals the pixel size (if there is
```

Comment:
This comment makes no sense. I think it should be 'the aggregated value equals the pixel _value_?